modules/axp/Animal/lib/Cow.pm

```diff
@@ -45,10 +49,11 @@ The default colour for this type of animal.
 =cut
 
 sub default_colour {
-	ref ( my $class = shift ) and croak "Static method used as instance call";
 	return "white and black";
 }
 
+before 'default_colour' => $LivingCreature::__static_check;
```

Comment:
Another hook, this time to make sure a class method isn't called on an instance. Yeah, Moose can assign a `before` to a method even if it didn't create the method itself. Nifty.